packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/rest_api/FlutterRestApi.kt

```diff
@@ -59,6 +65,18 @@ class FlutterRestApi {
             }
 
             try {
+                // Needed to prevent Android library from throwing a fatal error when body not present in some methods. https://github.com/aws-amplify/amplify-android/issues/1355
+                if (!options.hasData() && (methodName == PUT || methodName == POST || methodName == PATCH)) {
+                    handler.post {
+                        ExceptionUtil.postExceptionToFlutterChannel(flutterResult, "ApiException",
+                                ExceptionUtil.createSerializedError(
+                                        ApiException("$methodName request must have a body", "Add a body to the request.")
+                                )
+                        )
+                    }
+                    return
+                }
```

Comment:
Nit: could extract the logic into a throwable method similar to line 55-64, and indentations

Reply:
good suggestion, corrected in latest revisions